### DISCUSSION TITLE
Clearing abandoned cart

### DISCUSSION THREAD

**Discussion by ramkishore-yerubandi** [vendurehq/vendure#1270] [Q&A] [answered]
Is there any setting to release items from the Abandoned cart? Otherwise the available stock is blocked by long time opened state cart

**Comment by michaelbromley** [comment, marked as answer]
Stock does not by default get allocated when being just added to an order, only once the customer has checked out.

**Comment by amitavroy** [comment]
Well, I understand this is a question around how we can clear carts. 
But, can I understand how I can get a list of abandoned carts? 

For running certain marketing campaigns, it's quite common to look at what are some of the popular items in abandoned carts and hence I would like to know if there is a query available which I can run.

**Reply by michaelbromley** [reply]
Hi,
There's no built-in functionality for this, but you could [implement a custom query](https://www.vendure.io/docs/plugins/plugin-examples/extending-graphql-api/#adding-a-new-query-or-mutation) which does a DB query to find abandoned carts.

The actual DB query would be something like "find all orders where the most recent OrderLine.createdAt is older than <threshold>".

**Reply by amitavroy** [reply]
Ok, got it. I also expected this.. however a confirmation is better. Thanks for the reply. Appreciate it

**Reply by amitavroy** [reply]
Sorry to bother you again, but is it safe to say that any order item which is active = 1 and state = AddingItems then it's a cart which user is working through

**Reply by michaelbromley** [reply]
Yes, that's right. Even more concisely, `active: true` signifies it is the "active order", ie the one that will be affected by calls to `addItemToOrder` etc.
